### PR TITLE
Add shoppingcart model

### PR DESCRIPTION
**Pull request by denysokhrymovych**

### CODE REVIEW

File: src/main/java/com/example/bookshop/dto/cart/AddToCartRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import jakarta.validation.constraints.Positive;
+import lombok.Data;
+
+@Data
+public class AddToCartRequestDto {
+    @Positive(message = "Book id must be greater than 0")
+    private Long bookId;
+    @Positive(message = "Quantity must be greater than 0")
+    private int quantity;
```
Comment (yevheniisynytsia) [inline src/main/java/com/example/bookshop/dto/cart/AddToCartRequestDto.java:13]: please add `@NotNull` as well

File: src/main/java/com/example/bookshop/dto/cart/UpdateCartItemRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.example.bookshop.dto.cart;
+
+import jakarta.validation.constraints.Positive;
+import lombok.Data;
+
+@Data
+public class UpdateCartItemRequestDto {
+    @Positive(message = "Quantity must be greater than 0")
```
Comment (yevheniisynytsia) [inline src/main/java/com/example/bookshop/dto/cart/UpdateCartItemRequestDto.java:8]: same

File: src/main/java/com/example/bookshop/model/CartItem.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+@SQLDelete(sql = "UPDATE cart_items SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
+@Table(name = "cart_items")
+public class CartItem {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @ManyToOne
```
Comment (yevheniisynytsia) [inline src/main/java/com/example/bookshop/model/CartItem.java]: please avoid using `FetchType.EAGER`

File: src/main/java/com/example/bookshop/model/CartItem.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    @JoinColumn(name = "shopping_cart_id")
+    private ShoppingCart shoppingCart;
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @ManyToOne
```
Comment (yevheniisynytsia) [inline src/main/java/com/example/bookshop/model/CartItem.java]: please avoid using `FetchType.EAGER`

File: src/main/java/com/example/bookshop/model/ShoppingCart.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+@SQLDelete(sql = "UPDATE shopping_carts SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
+@Table(name = "shopping_carts")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @OneToOne
```
Comment (yevheniisynytsia) [inline src/main/java/com/example/bookshop/model/ShoppingCart.java]: please avoid using `FetchType.EAGER`

**Review by yevheniisynytsia — CHANGES_REQUESTED** [review]
Well done!
Left some comments to work on =)

File: src/main/java/com/example/bookshop/dto/cart/AddToCartRequestDto.java (hunk 1 above)
Comment (d1sam) [reply src/main/java/com/example/bookshop/dto/cart/AddToCartRequestDto.java:13]: still exist

File: src/main/java/com/example/bookshop/dto/cart/UpdateCartItemRequestDto.java (hunk 2 above)
Comment (d1sam) [reply src/main/java/com/example/bookshop/dto/cart/UpdateCartItemRequestDto.java:8]: add `@NotNull`

File: src/main/java/com/example/bookshop/service/shoppingcart/ShoppingCartServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final BookRepository bookRepository;
+    private final UserRepository userRepository;
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+
+    @Override
+    @Transactional
+    public ShoppingCartDto addToCart(AddToCartRequestDto requestDto, Long userId) {
```
Comment (d1sam) [inline src/main/java/com/example/bookshop/service/shoppingcart/ShoppingCartServiceImpl.java:31]: split to smaller private methods

File: src/main/java/com/example/bookshop/service/shoppingcart/ShoppingCartServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+    @Override
+    public ShoppingCartDto getCartByUserId(Long userId) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Shopping cart not found by user id: " + userId));
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
```
Comment (d1sam) [inline src/main/java/com/example/bookshop/service/shoppingcart/ShoppingCartServiceImpl.java:53]:
```suggestion
    @Transactional
    @Override
```

**Review by d1sam — CHANGES_REQUESTED** [review]
some mistakes still exist.

File: src/main/java/com/example/bookshop/dto/cart/AddToCartRequestDto.java (hunk 1 above)
Comment (yevheniisynytsia) [reply src/main/java/com/example/bookshop/dto/cart/AddToCartRequestDto.java:13]: my bad

@denysokhrymovych good catch about primitives and null values!

**Review by yevheniisynytsia — APPROVED** [review]
well done!

**Pull request merged**